src/description.cpp

```diff
@@ -339,6 +327,20 @@ string Description::generateSdp(string_view eol) const {
 	for (const auto &entry : mEntries) {
 		sdp << entry->generateSdp(eol, addr, port);
 
+		// RFC 8829: Attributes that SDP permits to be at either the session level or the media level
+		// SHOULD generally be at the media level even if they are identical.
+		sdp << "a=setup:" << mRole << eol;
+		if (mIceUfrag)
+			sdp << "a=ice-ufrag:" << *mIceUfrag << eol;
+		if (mIcePwd)
+			sdp << "a=ice-pwd:" << *mIcePwd << eol;
+		if (!mIceOptions.empty())
+			sdp << "a=ice-options:" << utils::implode(mIceOptions, ',') << eol;
```

Comment:
I see in RFC 8839, in section 5.6 it states `The "ice-options" attribute is a session-level and media-level attribute`. I'm not sure if another RFC overrules this, but if Firefox is keeping this session level it makes sense to leave it alone.

Reply:
It was a session-level attribute only in the older [RFC5245](https://datatracker.ietf.org/doc/html/rfc5245#section-15.5), indeed it changed in RFC 8839. Thank you for noticing, it looks like I wasn't up-to date :sweat_smile: I think it's still a argument in favor of letting it at session-level, in case some implementation of the older spec is picky about this.